analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -80,18 +88,61 @@ private fun getConcludedLicense(pkg: SpdxPackage): SpdxExpression? =
     pkg.licenseConcluded.takeIf { SpdxConstants.isPresent(it) }?.toSpdx()
 
 /**
- * Return a [RemoteArtifact] created from the downloadLocation of the given package [SpdxPackage] if it is a local
- * file, or return an [RemoteArtifact.EMPTY].
- *
- * TODO: Consider also taking "sourceInfo" or "externalRefs" into account.
+ * Return a [RemoteArtifact] for the binary artifact that the [pkg]'s [downloadLocation][SpdxPackage.downloadLocation]
+ * points to. If the download location is a "not present" value, or if it points to a source artifact or a VCS location
+ * instead, return null.
  */
-private fun getSourceArtifact(pkg: SpdxPackage): RemoteArtifact =
-    if (pkg.downloadLocation.startsWith("file:/")) {
-        RemoteArtifact(pkg.downloadLocation, Hash.NONE)
-    } else {
-        RemoteArtifact.EMPTY
+internal fun getBinaryArtifact(pkg: SpdxPackage): RemoteArtifact? =
+    getRemoteArtifact(pkg).takeUnless {
+        // Note: The "FilesAnalyzed" field "Indicates whether the file content of this package has been available for or
+        // subjected to analysis when creating the SPDX document". It does not indicate whether files were actually
+        // analyzed.
+        // If files were available, do *not* consider this do be a *binary* artifact.
+        pkg.filesAnalyzed
+    }
+
+/**
+ * Return a [RemoteArtifact] for the source artifact that the [pkg]'s [downloadLocation][SpdxPackage.downloadLocation]
+ * points to. If the download location is a "not present" value, or if it points to a binary artifact or a VCS location
+ * instead, return null.
+ */
+internal fun getSourceArtifact(pkg: SpdxPackage): RemoteArtifact? =
+    getRemoteArtifact(pkg).takeIf {
+        // Note: The "FilesAnalyzed" field "Indicates whether the file content of this package has been available for or
+        // subjected to analysis when creating the SPDX document". It does not indicate whether files were actually
+        // analyzed.
+        // If files were available, *do* consider this do be a *source* artifact.
+        pkg.filesAnalyzed
+    }
+
+private fun getRemoteArtifact(pkg: SpdxPackage): RemoteArtifact? =
+    when {
+        SpdxConstants.isNotPresent(pkg.downloadLocation) -> null
+        SPDX_VCS_PREFIXES.any { (prefix, _) -> pkg.downloadLocation.startsWith(prefix) } -> null
+        else -> RemoteArtifact(pkg.downloadLocation, Hash.NONE)
     }
 
+/**
+ * Return the [VcsInfo] contained in [pkg]'s [downloadLocation][SpdxPackage.downloadLocation], or null if the download
+ * location is a "not present" value / does not point to a VCS location.
+ */
+internal fun getVcsInfo(pkg: SpdxPackage): VcsInfo? =
+    SPDX_VCS_PREFIXES.mapNotNull { (prefix, vcsType) ->
+        // If the download location is a "not present" value, then none of the prefixes match, the mapping is empty, and
```

Comment:
It would be preferable to explicitly check the "not present" value, to prevent silently digesting malformed SPDX files.

Reply:
Done.